scripts/evaluate_model.py

```diff
@@ -0,0 +1,126 @@
+import argparse
+
+from datasets import load_dataset
+
+import numpy as np
+
+import torch
+from torch.utils.data import Dataset, DataLoader
+from torchtext.data.metrics import bleu_score
+
+from tqdm import tqdm
+
+from transformers import AutoTokenizer, AutoModelForSeq2SeqLM
+
+
+class ConvAI2CandidatesDataset(Dataset):
+    """ConvAI2 Candidates Dataset"""
+
+    def __init__(self, dataset):
+        """
+        Args:
+            dataset (DatasetDict): ConvAI2 dataset from huggingface
+        """
+
+        np.random.seed(42)
+        candidates_dataset = {"context": [], "candidates": []}
+        ids = list(range(len(dataset["train"])))
+
+        for i, dialog in (pbar := tqdm(enumerate(dataset["train"]))):
```

Comment:
This cycle is too complex, let's split it into a few functions. For example, extracting samples from one dialog may be moved into a separate function.